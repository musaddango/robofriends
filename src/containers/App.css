/* #### Generated By: http://www.cufonfonts.com #### */

@font-face {
font-family: 'SEGA LOGO FONT';
font-style: normal;
font-weight: normal;
src: local('SEGA LOGO FONT'), url('SEGA.woff') format('woff');
}

h1 {
  font-family: 'SEGA LOGO FONT';
  font-weight: 200;
  color: #0ccac4;
}

.scroll{overflow:scroll; 
/*    height:500px; */
/*    border:1px solid gray;*/
    }
.scroll::-webkit-scrollbar {
      display: none;
    }