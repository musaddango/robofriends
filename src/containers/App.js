import React, { Component } from 'react';
import { connect } from 'react-redux';
import MainPage from '../components/MainPage.js';
import { requestRobots, setSearchField } from '../redux/actions';
import './App.css';


const mapStateToProp = (state)=>({
  searchField: state.searchRobots.searchField,
  robots: state.requestRobots.robots,
  isPending: state.requestRobots.isPending,
  });
 
const mapDispatchToProp = (dispatch) =>({
  onSearchChange: (event)=> dispatch(setSearchField(event.target.value)),
  requestRobots: ()=> dispatch(requestRobots())
})


class App extends Component {

  render() {
      return(
        <MainPage {...this.props}/>
      )
  }
}

export default connect(mapStateToProp, mapDispatchToProp)(App);