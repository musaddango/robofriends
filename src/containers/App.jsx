import CardList from '../Components/cardList';
import SearchBox from '../Components/SearchBox';
import {robots} from '../Components/robots.js';
import {Component, useState, useEffect} from 'react';
import Scroll from '../Components/Scroll';
import ErrorBoundry from '../Components/ErrorBoundry';
import './App.css';
import 'tachyons';


// Functional Method using react hooks

const App = () => {
	

	const [robot, setRobots] = useState([]);
	const [searchField, setSearchField] = useState("");

	useEffect(()=> {
		fetch('https://jsonplaceholder.typicode.com/users')
		.then((response)=> response.json())
		.then((users)=> setRobots(users));
		// setRobots(robots)

	}, [])

	const onSearchChange = event=>{
		setSearchField(event.target.value);
	}

	
		const filteredRobots = robots.filter(robots =>{
			return robots.name.toLowerCase().includes(searchField.toLowerCase());
		});

		console.log(robots, searchField);

		//if robots.length === 0 (it gives false) if robots.length !== 0 (it gives true) !robots.length gives the opposite of both scenarios.
		return !robots.length ? <h1 className='tc'>Loading</h1>: 
		(
			<div className='tc scroll'>
				<div className='Scroll'>
					<h1 className='tc f1 sega'>RoboFriends</h1>
					<SearchBox searchChange= {onSearchChange} />
				</div>
				<ErrorBoundry>
					<Scroll>
						< CardList robots={filteredRobots}/>
					</Scroll>
				</ErrorBoundry>
			</div>
		);
	}


// 	// Class Component Method

// class App extends Component {
// 	constructor(){
// 		super()
// 		this.state = {
// 			robots: [],
// 			searchField:''
// 		} 
// 	}

// 	componentDidMount(){
// 		this.setState({robots: robots})
// 		// fetch('https://jsonplaceholder.typicode.com/users')
// 		// .then((response)=> response.json())
// 		// .then((users)=> this.setState({robots:users}))
// 	}

// 	onSearchChange = event=>{
// 		this.setState({searchField:event.target.value});
// 	}

// 	render(){
// 		const { robots, searchField } = this.state;
// 		const filteredRobots = robots.filter(robots =>{
// 			return robots.name.toLowerCase().includes(searchField.toLowerCase());
// 		});

// 		//if robots.length === 0 (it gives false) if robots.length !== 0 (it gives true) !robots.length gives the opposite of both scenarios.
// 		return !robots.length? <h1 className='tc'>Loading</h1>: 
// 		(
// 			<div className='tc scroll'>
// 				<div className='Scroll'>
// 					<h1 className='tc f1 sega'>RoboFriends</h1>
// 					<SearchBox searchChange= {this.onSearchChange} />
// 				</div>
// 				<ErrorBoundry>
// 				<Scroll>
// 						< CardList robots={filteredRobots}/>
// 				</Scroll>
// 				</ErrorBoundry>
// 			</div>
// 		);
// 	}
// }


export default App;
