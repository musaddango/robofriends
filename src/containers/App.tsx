import React, { Component } from 'react';
import { connect } from 'react-redux';
import MainPage from '../components/MainPage.tsx';
import { requestRobots, setSearchField } from '../redux/actions.js';
import './App.css';

interface mapState {
  searchField: string,
  robots: RobotObject[],
  isPending: boolean,
}

const mapStateToProp = (state)=>({
  searchField: state.searchRobots.searchField,
  robots: state.requestRobots.robots,
  isPending: state.requestRobots.isPending,
  });
 
const mapDispatchToProp = (dispatch) =>({
  onSearchChange: (event) => dispatch(setSearchField(event.target.value)),
  requestRobots: () => dispatch(requestRobots())
})

interface RobotObject{
  id: number,
  name: string,
  username: string,
  email: String,
}
interface AppProps {
  searchField: string,
  robots: RobotObject[],
  isPending: boolean,
  onSearchChange: React.ChangeEventHandler<HTMLInputElement> | undefined,
  requestRobots: any,
}

class App extends Component<AppProps, {}> {
  constructor(props){
    super(props);
  }

  render() {
      return(
        <MainPage {...this.props} />
      )
  }
}

export default connect(mapStateToProp, mapDispatchToProp)(App);