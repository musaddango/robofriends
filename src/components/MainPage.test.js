import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './MainPage';
import { robots } from './robots';

let wrapper;
beforeEach(()=>{
    const mockState = {
    robots,
    searchField: 'A',
    requestRobots: jest.fn()
}
    wrapper = shallow(<App 
    robots = {mockState.robots} 
    searchField = {mockState.searchField} 
    requestRobots={mockState.requestRobots}/>)
})

it('It renders main page without crashing.', ()=>{
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('Robots are properly filtered.', ()=>{
    
})