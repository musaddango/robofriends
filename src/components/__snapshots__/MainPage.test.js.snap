// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders main page without crashing. 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={
          Array [
            Object {
              "email": "Sincere@april.biz",
              "id": 1,
              "name": "Leanne Graham",
              "username": "Bret",
            },
            Object {
              "email": "Nathan@yesenia.net",
              "id": 3,
              "name": "Clementine Bauch",
              "username": "Samantsha",
            },
            Object {
              "email": "Julianne.OConner@kory.org",
              "id": 4,
              "name": "Patricia Lebsack",
              "username": "Karianne",
            },
            Object {
              "email": "Telly.Hoeger@billy.biz",
              "id": 7,
              "name": "Kurtis Weissnat",
              "username": "Elwyn.Skiles",
            },
            Object {
              "email": "Sherwood@rosamond.me",
              "id": 8,
              "name": "Nicholas Runolfsdottir V",
              "username": "Maxime_Nienow",
            },
            Object {
              "email": "Chaim_McDermott@dana.io",
              "id": 9,
              "name": "Glenna Reichert",
              "username": "Delphine",
            },
            Object {
              "email": "Rey.Padberg@karina.biz",
              "id": 10,
              "name": "Clementina DuBuque",
              "username": "Moriah.Stanton",
            },
          ]
        }
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
